File: java/src/main/java/Book.java
public class Book {

    public enum Type {HARDBACK, PAPERBACK};
    
    private final String title;
    private final String author;
    private double recomendedRetailPrice;

    public double getRecomendedRetailPrice() {
        return recomendedRetailPrice;
    }
    public void setRecomendedRetailPrice(double recomendedRetailPrice) {
        this.recomendedRetailPrice = recomendedRetailPrice;
    }
    private Type type;
    private int monthsSincePublication;

    public Type getType() {
        return type;
    }
    public void setType(Type type) {
        this.type = type;
    }
    public int getMonthsSincePublication() {
        return monthsSincePublication;
    }
    public void setMonthsSincePublication(int monthsSincePublication) {
        this.monthsSincePublication = monthsSincePublication;
    }
    public Book(String title, String author) {
        this.title = title;
        this.author = author;
    }
    public String getTitle() {
        return title;
    }
    public String getAuthor() {
        return author;
    }

}

File: typescript/src/Book.ts
export enum BookType {
    HARDBACK,
    PAPERBACK
}

export class Book {
    private title: string;
    private author: string;
    private recommendedRetailPrice: number;
    private type: BookType;
    private monthsSincePublication: number;

    constructor(title: string, author: string) {
        this.title = title;
        this.author = author;
    }

    public getRecommendedRetailPrice(): number {
        return this.recommendedRetailPrice;
    }

    public setRecommendedRetailPrice(recommendedRetailPrice: number): void {
        this.recommendedRetailPrice = recommendedRetailPrice;
    }

    public getType(): BookType {
        return this.type;
    }

    public setType(type: BookType): void {
        this.type = type;
    }

    public getMonthsSincePublication(): number {
        return this.monthsSincePublication;
    }

    public setMonthsSincePublication(monthsSincePublication: number): void {
        this.monthsSincePublication = monthsSincePublication;
    }

    public getTitle(): string {
        return this.title;
    }

    public getAuthor(): string {
        return this.author;
    }
}
